src/Type/Php/RoundFunctionReturnTypeExtension.php

```diff
@@ -39,21 +38,18 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 		);
 	}
 
-	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
 	{
+		// PHP 7 can return either a float or false.
+		// PHP 8 can either return a float or fatal.
+		$defaultReturnType = null;
+
 		if ($this->phpVersion->hasStricterRoundFunctions()) {
 			// PHP 8 fatals with a missing parameter.
 			$noArgsReturnType = new NeverType(true);
-			// PHP 8 can either return a float or fatal.
-			$defaultReturnType = new FloatType();
 		} else {
 			// PHP 7 returns null with a missing parameter.
 			$noArgsReturnType = new NullType();
-			// PHP 7 can return either a float or false.
-			$defaultReturnType = new BenevolentUnionType([
-				new FloatType(),
-				new ConstantBooleanType(false),
-			]);
```

Comment:
This extension change is unnecessary, but it removes the double management of types with functionMap.

Reply:
just stumbled over https://github.com/phpstan/phpstan-src/blob/cbdc933b68f42dacf44143441595b9df36ce7f2b/src/Type/Php/StrtotimeFunctionReturnTypeExtension.php#L39-L41 which might benefit from a similar simplification.